test/plots/plotTests.ts

```diff
@@ -1,584 +1,302 @@
-
-      let entities = plot.entities();
-      assert.lengthOf(entities, data1.length + data2.length, "retrieved one Entity for each value on the Plot");
-      let entityData = entities.map((entity) => entity.datum);
-      assert.includeMembers(entityData, data1, "includes data1 members");
-      assert.includeMembers(entityData, data2, "includes data2 members");
-      let entityPositions = entities.map((entity) => entity.position);
-      assert.includeMembers(entityPositions, data1.map(data1PointConverter), "includes data1 points");
-      assert.includeMembers(entityPositions, data2.map(data2PointConverter), "includes data2 points");
-
-      entities = plot.entities([dataset1]);
-      assert.lengthOf(entities, data1.length, "retrieved one Entity for each value in dataset1");
-      assert.strictEqual(entities[0].selection.node(), renderArea1Selection.node(), "returns the selection associated with dataset1");
-      assert.includeMembers(entities.map((entity) => entity.datum), data1, "includes data1 members");
-      assert.includeMembers(entities.map((entity) => entity.position), data1.map(data1PointConverter), "includes data1 points");
-
-      entities = plot.entities([dataset2]);
-      assert.lengthOf(entities, data2.length, "retrieved one Entity for each value in dataset2");
-      assert.strictEqual(entities[0].selection.node(), renderArea2Selection.node(), "returns the selection associated with dataset1");
-      assert.includeMembers(entities.map((entity) => entity.datum), data2, "includes data1 members");
-      assert.includeMembers(entities.map((entity) => entity.position), data2.map(data2PointConverter), "includes data2 points");
-      svg.remove();
-    });
+        plot.removeDataset(dataset);
 
-    it("entities() with NaN values", () => {
-      let svg = TestMethods.generateSVG(400, 400);
-      let plot = new Plottable.Plot();
+        assert.deepEqual(scale.domain(), oldDomain, "domain does not include dataset if removed");
 
-      let data = [{value: NaN}, {value: 1}, {value: 2}];
-      let dataset = new Plottable.Dataset(data);
+        const data2 = [7, -7, 8];
+        const dataset2 = new Plottable.Dataset(data2);
+        plot.datasets([dataset, dataset2]);
 
-      let dataPoints = data.map((datum: any) => { return {x: datum.value, y: 10}; });
+        assert.operator(scale.domainMin(), "<=", Math.min.apply(null, data.concat(data2)), "domainMin includes new dataset");
+        assert.operator(scale.domainMax(), ">=", Math.max.apply(null, data.concat(data2)), "domainMax includes new dataset");
 
-      let dataPointConverter = (datum: any, index: number) => dataPoints[index];
+        svg.remove();
+      });
 
-      // Create mock drawer with already drawn items
-      let mockDrawer = new Plottable.Drawer(dataset);
-      (<any> mockDrawer)._svgElementName = "circle";
-      mockDrawer.selector = () => "circle";
+      it("updates the scale extents in dataset order", () => {
```

Comment:
What I'm getting that is that for most scales, the order of extent updating isn't relevant.